zipline/utils/cme_exchange_calendar.py

```diff
@@ -0,0 +1,454 @@
+    day=1,
+    # When Jan 1 is a Sunday, NYSE observes the subsequent Monday.  When Jan 1
+    # Saturday (as in 2005 and 2011), no holiday is observed.
+    observance=sunday_to_monday
+)
+USMemorialDay = Holiday(
+    # NOTE: The definition for Memorial Day is incorrect as of pandas 0.16.0.
+    # See https://github.com/pydata/pandas/issues/9760.
+    'Memorial Day',
+    month=5,
+    day=25,
+    offset=DateOffset(weekday=MO(1)),
+)
+USMartinLutherKingJrAfter1998 = Holiday(
+    'Dr. Martin Luther King Jr. Day',
+    month=1,
+    day=1,
+    # The NYSE didn't observe MLK day as a holiday until 1998.
+    start_date=Timestamp('1998-01-01'),
+    offset=DateOffset(weekday=MO(3)),
+)
+USIndependenceDay = Holiday(
+    'July 4th',
+    month=7,
+    day=4,
+    observance=nearest_workday,
+)
+Christmas = Holiday(
+    'Christmas',
+    month=12,
+    day=25,
+    observance=nearest_workday,
+)
+
+# Half Days
+MonTuesThursBeforeIndependenceDay = Holiday(
+    # When July 4th is a Tuesday, Wednesday, or Friday, the previous day is a
+    # half day.
+    'Mondays, Tuesdays, and Thursdays Before Independence Day',
+    month=7,
+    day=3,
+    days_of_week=(MONDAY, TUESDAY, THURSDAY),
+    start_date=Timestamp("1995-01-01"),
+)
+FridayAfterIndependenceDayExcept2013 = Holiday(
+    # When July 4th is a Thursday, the next day is a half day (except in 2013,
+    # when, for no explicable reason, Wednesday was a half day instead).
+    "Fridays after Independence Day that aren't in 2013",
+    month=7,
+    day=5,
+    days_of_week=(FRIDAY,),
+    observance=lambda dt: None if dt.year == 2013 else dt,
+    start_date=Timestamp("1995-01-01"),
+)
+USBlackFridayBefore1993 = Holiday(
+    'Black Friday',
+    month=11,
+    day=1,
+    # Black Friday was not observed until 1992.
+    start_date=Timestamp('1992-01-01'),
+    end_date=Timestamp('1993-01-01'),
+    offset=[DateOffset(weekday=TH(4)), Day(1)],
+)
+USBlackFridayInOrAfter1993 = Holiday(
+    'Black Friday',
+    month=11,
+    day=1,
+    start_date=Timestamp('1993-01-01'),
+    offset=[DateOffset(weekday=TH(4)), Day(1)],
+)
+# These have the same definition, but are used in different places because the
+# NYSE closed at 2:00 PM on Christmas Eve until 1993.
+ChristmasEveBefore1993 = Holiday(
+    'Christmas Eve',
+    month=12,
+    day=24,
+    end_date=Timestamp('1993-01-01'),
+    # When Christmas is a Saturday, the 24th is a full holiday.
+    days_of_week=(MONDAY, TUESDAY, WEDNESDAY, THURSDAY),
+)
+ChristmasEveInOrAfter1993 = Holiday(
+    'Christmas Eve',
+    month=12,
+    day=24,
+    start_date=Timestamp('1993-01-01'),
+    # When Christmas is a Saturday, the 24th is a full holiday.
+    days_of_week=(MONDAY, TUESDAY, WEDNESDAY, THURSDAY),
+)
+
+
+# http://en.wikipedia.org/wiki/Aftermath_of_the_September_11_attacks
+September11Closings = date_range('2001-09-11', '2001-09-16', tz='UTC')
+
+# http://en.wikipedia.org/wiki/Hurricane_sandy
+HurricaneSandyClosings = date_range(
```

Comment:
I previously thought it was, but more digging revealed that trading was just 'disrupted', so the code as it stands is out of date. See this for more info: https://github.com/quantopian/dataloader/blob/futures-exp-dates/dataloader/resources/tradingcalendar_cme.py#L254-L274